### PR TITLE
Hand Restraint Refactor

### PR DESCRIPTION
**Pull request by Shadowmech88**
Is refactor the right word? I'm not sure.
Anyway, this PR moves hand restraining from `/obj/item/weapon/handcuffs` all the way up to `/obj/item`.
Now any item with its `restraint_resist_time` var set can be used to restrain a mob's hands. In most cases, this will require being in grab intent. Handcuffs/cablecuffs can still be used to restrain a mob while in any intent.
As a demonstration of the new capability, belts can now be used to restrain a mob's hands, having the same resist time as cablecuffs do.

The goal with this PR is to make truly improvised restraints more possible, and so hopefully cut down on the need people feel to powergame handcuffs/cablecuffs.

:cl:
 * rscadd: Belts can now be used to restrain a player's hands. To do so, simply attack the player with the belt while in grab intent.

### CODE REVIEW

**Comment by Intigracy** [comment]
Belts should be better than cablecuffs, there's less of them.

**Comment by Blithering** [comment]
I'd argue against them being better, as cablecuffs are contraband, anyone caught carrying some will be outed as planning to abduct someone, and making them as and when they're needed takes up time your victim could use to get away. Belts on the other hand are innocuous enough to pass a search, so that's already a plus. 
I do want to see spaghetti restraints with a 1 second break time though

**Comment by ghost** [comment]
OH NO
THE PLASTIC​ BAG NERF
WE KNEW IT WOULD COME SOMEDAY

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -1127,3 +1130,72 @@ var/global/list/image/blood_overlays = list()
...
+			to_chat(user, "<span class='warning'>Uh... how is this done?!</span>")
+			C = user
+
+		user.visible_message("<span class='danger'>\The [user] has restrained \the [C] with \the [src]!</span>")
+		user.attack_log += text("\[[time_stamp()]\] <font color='red'>Has restrained [C.name] ([C.ckey]) with \the [src].</font>")
+		C.attack_log += text("\[[time_stamp()]\] <font color='red'>Restrained with \the [src] by [user.name] ([user.ckey])</font>")
+		log_attack("[user.name] ([user.ckey]) has restrained [C.name] ([C.ckey]) with \the [src]")
+
+		var/obj/item/cuffs = src
+		if(istype(src, /obj/item/weapon/handcuffs/cyborg)) //There's GOT to be a better way to check for this.
```
Comment (SonixApache) [inline code/game/objects/items.dm:1190]: could make a helper proc, but that's it

**Comment by SonixApache** [comment]
@Shadowmech88 tested? Good to go? Page doctor inti if it is.

**Comment by Shadowmech88** [comment]
@Intigracy this is tested and good to go.

**Pull request merged**